lib/dspace-client.ts

```diff
@@ -0,0 +1,193 @@
+// @ts-check
+import got, { HTTPError } from 'got';
+import { Parser } from 'xml2js';
+
+import { DSpaceItem, Metadata, RSSFeed } from './dspace-schemas';
+
+const headers = {
+  Accept: 'application/json',
+  'Content-Type': 'application/json',
+  'User-Agent': 'Mozilla/5.0 (iPhone; U; CPU iPhone OS 4_3_3 like Mac OS X; en-us) AppleWebKit/533.17.9 (KHTML, like Gecko) Version/5.0.2 Mobile/8J2 Safari/6533.18.5',
+};
+
+/**
+ * Searches for DSpace items that match the given metadata field and value.
+ * Includes the metadata and bitstreams fields for each matched item.
+ *
+ * @param endpoint - DSpace endpoint. Should include protocol.
+ * @param key - Metadata field key
+ * @param value - Metadata field value
+ * @returns List of DSpace items that match the query.
+ */
+export const find = async (
+  endpoint: string,
+  key: string,
+  value: string
+): Promise<DSpaceItem[]> => got.post(`${endpoint}/rest/items/find-by-metadata-field`, {
+  json: {
+    key,
+    value,
+  },
+  searchParams: {
+    expand: 'metadata,bitstreams',
+  },
+  responseType: 'json',
+  resolveBodyOnly: true,
+  headers,
+});
+
+/**
+ * Fetches the DSpace RSS feed.
+ *
+ * @param endpoint - DSpace endpoint. Should include protocol.
+ * @returns Parsed DSpace RSS feed.
+ */
+export const getFeed = async (endpoint: string): Promise<RSSFeed> => {
+  const rawFeedXML = await got(`${endpoint}/feed/rss_2.0/site`, {
+    headers: {
+      ...headers,
+      Accept: 'application/xml',
+    },
+    resolveBodyOnly: true,
+  });
+
+  return new Promise((resolve, reject) => {
```

Comment:
Parser supports Promises. This can be cleaned up to:

```ts
const parser = new Parser({ explicitRoot: false });
return parser.parseStringPromise(rawFeedXML);
  ```